cpp/order/SolidLiquid.h

```diff
@@ -8,7 +8,7 @@
 #include <iterator>
 #include <vector>
 
-#include "Cluster.h"
+#include "../cluster/Cluster.h"
```

Comment:
I think this is unnecessary.

Reply:
`freud/cpp/cluster` is not in the include search path. `clang-tidy` cannot find the file without this change:
https://github.com/glotzerlab/freud/actions/runs/9649688151/job/26613682932#step:6:67

Reply:
Okay, I suppose we can keep it for now and revert this change in a future PR converting the order module.